src/control/tools/StrokeStabilizer.cpp

```diff
@@ -0,0 +1,408 @@
+
+        lastLiveEvent = lastEvent;
+
+        resetBuffer(lastPaintedEvent, pos.timestamp);
+        return;
+    }
+
+    /**
+     * Normal behaviour. Adjust the event's coordinates according to the deadzoneRadius
+     */
+    lastLiveEvent = lastEvent;
+    lastLiveDirection = movement;
+
+    Event ev(lastEvent.x - ratio * movement.dx, lastEvent.y - ratio * movement.dy, lastEvent.pressure);
+
+    averageAndPaint(ev, pos.timestamp);
+}
+
+void StrokeStabilizer::Deadzone::rebalanceStrokePressures() {
+    Stroke* stroke = strokeHandler->getStroke();
+    int pointCount = stroke->getPointCount();
+    if (pointCount >= 3) {
+        /**
+         * Smoothen a little bit the pressure variations
+         */
+        stroke->setSecondToLastPressure((stroke->getPoint(pointCount - 2).z + stroke->getPoint(pointCount - 3).z) / 2);
+    }
+}
+
+
+/**
+ * StrokeStabilizer::Inertia
+ */
+void StrokeStabilizer::Inertia::recordFirstEvent(const PositionInputData& pos) {
+    lastEvent = Event(pos);
+    lastPaintedEvent = lastEvent;
+}
+
+void StrokeStabilizer::Inertia::processEvent(const PositionInputData& pos) {
+
+    /**
+     * Record the event for the stroke finisher
+     */
+    lastEvent = Event(pos);
+
+    /**
+     * Compute the acceleration due to the spring action
+     */
+    MathVect springAcceleration = {(lastEvent.x - lastPaintedEvent.x) / mass,
+                                   (lastEvent.y - lastPaintedEvent.y) / mass};
+
+    speed.dx = speed.dx * oneMinusDrag + springAcceleration.dx;
+    speed.dy = speed.dy * oneMinusDrag + springAcceleration.dy;
+
+    Event ev(lastPaintedEvent.x + speed.dx, lastPaintedEvent.y + speed.dy, lastEvent.pressure);
+
+    averageAndPaint(ev, pos.timestamp);
+}
+
+void StrokeStabilizer::Inertia::rebalanceStrokePressures() {
+    Stroke* stroke = strokeHandler->getStroke();
+    int pointCount = stroke->getPointCount();
+    if (pointCount >= 3) {
+        /**
+         * Smoothen a little bit the pressure variations
+         */
+        stroke->setSecondToLastPressure((stroke->getPoint(pointCount - 2).z + stroke->getPoint(pointCount - 3).z) / 2);
+    }
+}
+
+
+/**
+ * StrokeStabilizer::Arithmetic
+ */
+void StrokeStabilizer::Arithmetic::recordFirstEvent(const PositionInputData& pos) {
+    eventBuffer.assign(Event(pos));  // Fill the buffer with copies of Event(pos)
+}
+
+void StrokeStabilizer::Arithmetic::averageAndPaint(const Event& ev, guint32 timestamp) {
+    /**
+     * Push the event and overwrite the oldest event in the buffer
+     */
+    eventBuffer.push_front(ev);
+
+    /**
+     * Average the coordinates using an arithmetic mean
+     */
+    Event sum = std::accumulate(begin(eventBuffer), end(eventBuffer), Event(0, 0, 0), [](auto&& lhs, auto&& rhs) {
+        return Event(lhs.x + rhs.x, lhs.y + rhs.y, lhs.pressure + rhs.pressure);
+    });
+
+    /**
+     * Rescale the averaged coordinates and draw
+     */
+    double d = (double)eventBuffer.size();
```

Comment:
do not use c-style casts:

```suggestion
    double d = static_cast<double>(eventBuffer.size());
```